internal/oci/store/overlay/store_overlay.go

```diff
@@ -173,8 +173,11 @@ func (c *CachingBundler) Bundle(ctx context.Context, i ociv1.Image, id string, o
 	if err != nil {
 		return nil, errors.Wrap(err, errGetLayers)
 	}
-
-	lowerPaths := make([]string, len(layers))
+	nLayers := len(layers)
+	if nLayers > store.MaxLayers {
+		return nil, errors.Errorf(store.ErrFmtTooManyLayers, nLayers, store.MaxLayers)
+	}
```

Comment:
Is it worth making this a function in the `oci` package? We could then export and call that function rather than exporting a const error string. Something like `oci.Validate(img)`.

We could also contribute this to go-container registry, which already has a `validate.Image(img)` function. I could imagine something like `validate.Image(img, validate.MaxLayers(256))` being useful for other if this is indeed a general security concern.

Reply:
See https://github.com/crossplane/crossplane/pull/4203/commits/7fd862047d6aa81d1811214db55fde3966bf26ed, I couldn't move it to `oci` because I would have needed to either export or duplicate `errGetLayers`, which we could definitely do, but doesn't feel right. The main drawback I can see is that we have to call `Layers` twice, which is necessary if we keep the `Image` as input.